### PR TITLE
Allow user to provide Top List file and icon's minimum resolution

### LINKED ISSUE #23: Customize tippy top to accept Top List file and icon's minimum resolution from user
The script currently works:

1. Only with [Alexa top sites](https://github.com/mozilla/tippy-top-sites/blob/master/make_manifest.py#L20) and
2. Returns icons only of there is one that is at least [96 pixel wide](https://github.com/mozilla/tippy-top-sites/blob/master/make_manifest.py#L124)

 
It would be useful if we could allow users to specify a top sites file (e.g. a csv file) and minimum resolution. This can greatly increase the re-usability of this script (https://mozilla-hub.atlassian.net/browse/DENG-407 is one of the examples).

I propose to add these 2 as cli options. Both of these options will be optional. If any of them is not specified, the script will work as before i.e. with Alexa top sites and 96 pixel width to be backward compatible.

### PR DESCRIPTION
**Pull request by abhi-agg**
Fixes https://github.com/mozilla/tippy-top-sites/issues/23

### CODE REVIEW

**Comment by abhi-agg** [comment]
@ncloudioj Could you please review it? I don't have the permission to add you as a reviewer in this repo. Thanks :)

File: make_manifest.py [hunk 1]
```diff
@@ -49,12 +50,23 @@ def _fetch_alexa_top_sites():
...
+                continue
+            yield (row[0], row[1])
 
-def alexa_top_sites(count=1000):
-    logging.info('Fetching Alexa top {count} sites'.format(count=count))
-    top = _fetch_alexa_top_sites()
-    return [next(top) for x in range(count)]
-
+def top_sites(topsitesfile, count):
+    logging.info('Fetching top {count} sites'.format(count=count))
```
Comment (ncloudioj) [inline make_manifest.py]:
```suggestion
    logging.info(f'Fetching top {count} sites')
```

File: make_manifest.py [hunk 2]
```diff
@@ -141,30 +152,33 @@ def collect_icons_for_alexa_top(count, extra_domains=None):
...
+            'best_icon': best_icon_url
         })
     logging.info('Done fetching icons')
     return results
 
 
 @click.command()
-@click.option('--count', default=10, help='Number of sites from Alexa Top Sites')
+@click.option('--count', default=10, help='Number of sites from a list of Top Sites that should be used to generate the manifest. Default is 10.')
+@click.option('--topsitesfile', type=click.Path(exists=True), help='A csv file containing comma separated rank and domain information (in the same order) of the Top Sites. If no file is provided then Alexa Top Sites are used.')
```
Comment (ncloudioj) [inline make_manifest.py:169]: Shall we make this argument optional? Since it will fall back to Alexa if not specified.

Comment (abhi-agg) [reply make_manifest.py:169]: The option `--topsitesfile` is optional in the sense if the user doesn't specify it on command line, it is not used. Is this what you meant?

File: make_manifest.py (hunk 1 above)
Comment (abhi-agg) [reply make_manifest.py]: Done.

File: make_manifest.py (hunk 2 above)
Comment (ncloudioj) [reply make_manifest.py:169]: Yep, I didn't see a default value here, also noticed the path check, so just wanted to make sure it's ok to omit this argument.

**Review by ncloudioj — APPROVED** [review]
Looks good, thanks!

**Pull request merged**